src/localization/finnish.ts

```diff
@@ -25,61 +25,64 @@ export var finnishSurveyStrings = {
+  maxSelectError: "Valitse enintään {0} vaihtoehtoa.",
   numericMinMax:
-    "'{0}' täytyy olla enemmän tai yhtä suuri kuin {1} ja vähemmän tai yhtä suuri kuin {2}",
-  numericMin: "'{0}' täytyy olla enemmän tai yhtä suuri kuin {1}",
-  numericMax: "'{0}' täytyy olla vähemmän tai yhtä suuri kuin {1}",
+    "Luvun '{0}' tulee olla vähintään {1} ja korkeintaan {2}.",
+  numericMin: "Luvun '{0}' tulee olla vähintään {1}.",
+  numericMax: "Luvun '{0}' tulee olla korkeintaan {1}.",
   invalidEmail: "Syötä validi sähköpostiosoite.",
   invalidExpression: "Lausekkeen: {0} pitäisi palauttaa 'true'.",
   urlRequestError: "Pyyntö palautti virheen {0}. {1}",
   urlGetChoicesError:
-    "Pyyntö palautti tyhjät tiedot tai polku-ominaisuus on väärä",
+    "Pyyntö palautti tyhjän tiedoston tai 'path'-asetus on väärä",
   exceedMaxSize: "Tiedoston koko ei saa olla suurempi kuin {0}.",
-  otherRequiredError: "Anna toinen arvo.",
+  otherRequiredError: "Tarkenna vastaustasi tekstikenttään.",
   uploadingFile:
-    "Tiedostoa ladataan. Odota muutama sekunti ja yritä uudelleen.",
+    "Tiedostoa lähetetään. Odota muutama sekunti ja yritä uudelleen.",
   loadingFile: "Ladataan...",
-  chooseFile: "Valitse tiedosto (t) ...",
+  chooseFile: "Valitse tiedosto(t)...",
   noFileChosen: "Ei tiedostoa valittuna",
-  confirmDelete: "Haluatko poistaa tietueen?",
-  keyDuplicationError: "Tämän arvon tulisi olla ainutlaatuinen.",
+  confirmDelete: "Haluatko poistaa osion?",
+  keyDuplicationError: "Tämä arvo on jo käytössä. Syötä toinen arvo.",
   addColumn: "Lisää sarake",
   addRow: "Lisää rivi",
   removeRow: "Poista",
+  emptyRowsText: "Ei rivejä",
   addPanel: "Lisää uusi",
   removePanel: "Poista",
   choices_Item: "kohde",
   matrix_column: "Sarake",
   matrix_row: "Rivi",
-  savingData: "Tulokset tallennetaan palvelimelle ...",
-  savingDataError: "Tapahtui virhe, emmekä voineet tallentaa tuloksia.",
-  savingDataSuccess: "Tulokset tallennettiin onnistuneesti!",
+  savingData: "Tietoja tallennetaan palvelimelle...",
+  savingDataError: "Tapahtui virhe, emmekä voineet tallentaa kyselyn tietoja.",
+  savingDataSuccess: "Tiedot tallennettiin onnistuneesti!",
   saveAgainButton: "Yritä uudelleen",
   timerMin: "min",
   timerSec: "sek",
   timerSpentAll: "Olet käyttänyt {0} tällä sivulla ja yhteensä {1}.",
-  timerSpentPage: "Olet viettänyt {0} tällä sivulla.",
+  timerSpentPage: "Olet käyttänyt {0} tällä sivulla.",
   timerSpentSurvey: "Olet käyttänyt yhteensä {0}.",
   timerLimitAll:
     "Olet käyttänyt tällä sivulla {0} / {1} ja yhteensä {2} / {3}.",
   timerLimitPage: "Olet käyttänyt {0} / {1} tällä sivulla.",
   timerLimitSurvey: "Olet käyttänyt yhteensä {0} / {1}.",
-  cleanCaption: "Puhdas",
-  clearCaption: "Asia selvä",
+  cleanCaption: "Pyyhi",
+  clearCaption: "Tyhjennä",
   chooseFileCaption: "Valitse tiedosto",
   removeFileCaption: "Poista tämä tiedosto",
-  booleanCheckedLabel: "Joo",
+  booleanCheckedLabel: "Kyllä",
```

Comment:
_Kyllä_ is more formal than _joo_, which is basically equivalent to "yeah" or "yep" in English.